.github/workflows/cd.yml

```diff
@@ -68,23 +68,31 @@ jobs:
                   REDIS_URL: 'redis://localhost'
                   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
-            - name: Set LAST_CHANGELOG_ENTRY
-              id: set-last-changelog-entry
+            - name: Create GitHub release
+              env:
+                GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
               run: |
                 # read from the first until the second header in the changelog file
                 # this assumes the formatting of the file
                 # and that this workflow is always running for the most recent entry in the file 
-                awk -v defText='see CHANGELOG.md' '/^## /{if (flag) exit; flag=1} flag && /^##$/{exit} flag; END{if (!flag) print defText}' CHANGELOG.md
-                echo "LAST_CHANGELOG_ENTRY=$LAST_CHANGELOG_ENTRY" >> "$GITHUB_OUTPUT"
-
-            - name: Create GitHub release
-              uses: actions/create-release@v1
-              env:
-                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-              with:
-                  tag_name: v${{ env.COMMITTED_VERSION }}
-                  release_name: ${{ env.COMMITTED_VERSION }}
-                  body: ${{ steps.set-last-changelog-entry.outputs.LAST_CHANGELOG_ENTRY }}
+                LAST_CHANGELOG_ENTRY=$(awk -v defText="see CHANGELOG.md" '/^## /{if (flag) exit; flag=1} flag && /^##$/{exit} flag; END{if (!flag) print defText}' CHANGELOG.md)
+                # the action we used to use was archived, and made it really difficult to create a release with a body
+                # because the LAST_CHANGELOG_ENTRY contains bash special characters so passing it between steps
+                # was a pain.
+                # we can use the github cli to create a release with a body
+                # all as part of one step
+                gh api \
+                  --method POST \
+                  -H "Accept: application/vnd.github+json" \
+                  -H "X-GitHub-Api-Version: 2022-11-28" \
+                  /repos/posthog/posthog-js/releases \
+                  -f tag_name="${{ env.COMMITTED_VERSION }}" \
+                -f target_commitish='main' \
+                -f name="${{ env.COMMITTED_VERSION }}" \
+                -f body="$LAST_CHANGELOG_ENTRY" \
+                -F draft=false \
+                -F prerelease=false \
+                -F generate_release_notes=false 
```

Comment:
I think it'd be better to keep using a GH action such as https://github.com/softprops/action-gh-release that keeps always up to date.
I think `GITHUB_OUTPUT` was just across jobs but in this case we need across steps which is `GITHUB_ENV`
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable
Not sure if it was the issue but... just a hunch.
I've used https://github.com/softprops/action-gh-release in the past and it accepts the `body`.
Will approve anyway, just wanted to share the possible issue of reusing the env. var. across steps, and most likely the value of the var should be wrapped by `''` to be considered a single string

Reply:
> I think GITHUB_OUTPUT was just across jobs but in this case we need across steps which is GITHUB_ENV

neither of those could cope with the "special" characters in the output. 

you can base64 encode the value, but then I couldn't successfully decode it on the other end

in the end I opted to cut a dependency which is ultimately just wrapping this command

Reply:
(I also didn't want to spend too much time on this and it was turning into a rabbit hole 🙈 )

Reply:
totally understand, I already had this problem in the past, but I recall using `GITHUB_ENV` and escaping the var value worked, but yeah, the `gh` cli should be fine as well.